src/views/Releases/ListReleases/index.jsx

```diff
@@ -134,8 +143,14 @@ function ListReleases(props) {
               );
             }
 
-            // todo: set these
-            release.required_signoffs = {};
+            if (release.read_only && !requiresSignoffs(release)) {
```

Comment:
The previous version was overriding the required signoffs to `{}`, so I deleted the line `release.required_signoffs = {};`, so the signoffs that points to a rule, wil not be empty anymore.

So `release.read_only && !requiresSignoffs(release)` handles the case when Release is readonly and not pointed by a rule. By calling `fetchRequiredSignoffsForProduct` will produces a response with that logic:

![image](https://user-images.githubusercontent.com/9947646/67347893-cd5e2f00-f519-11e9-992d-ccfc1b3c62c5.png)

Reply:
Ah, I see. In that case, can you add a comment that adds an explanation like your second paragraph above? I think the need to override existing `required_signoffs` is not quite obvious, and we're bound to forget.

Reply:
About sucessive calls to `fetchRequiredSignoffsForProduct`, I not finished yet, I intended to ask on IRC  if it is OK to use the `useMemo` React hook for this case. 

What do you think?

Reply:
I don't think `useMemo` is going to work for this case. The way it works is that it is only called when one of the dependencies (the second argument to `useMemo`) changes. I can't imagine an argument that would allow it to be called in the correct circumstances - but if you can, go ahead.

What I was thinking was simply adding another piece of state, something like:
```const [releaseRequiredSignoffs, setReleaseRequiredSignoffs] = useState({});```

Which would by keyed on product name, and the values would be the value returned by `fetchRequiredSignoffsForProduct`. When we need required signoffs for a release, we would look to see if they exist in `releaseRequiredSignoffs`. If they do, just use them. If they don't, fetch them and add them to it.

Reply:
> I don't think `useMemo` is going to work for this case. The way it works is that it is only called when one of the dependencies (the second argument to `useMemo`) changes. I can't imagine an argument that would allow it to be called in the correct circumstances - but if you can, go ahead.

How for this case we get required signoffs for product, I thought to pass as second parameter `[release.product]`, so any other release related with that product will get the same reference on rendering time.

When I created the `requiredSignoffsForProduct` endpoint, the goal was infer product from release (so parameter is release name), once it is a specific release related behaviour.

I will try the state approach...

Reply:
Feel free to give `useMemo` a try with `release.product`!